src/dune_rules/melange_rules.ml

```diff
@@ -9,6 +9,33 @@ let make_js_name ~js_ext ~dst_dir m =
   let name = Melange.js_basename m ^ js_ext in
   Path.Build.relative dst_dir name
 
+let virtual_dst_dir_and_modules sctx ~target_dir ~implements ~scope =
+  let open Memo.O in
+  match implements with
+  | None -> Memo.return None
+  | Some (loc, implements) -> (
+    Lib.DB.find (Scope.libs scope) implements >>= function
+    | None ->
+      User_error.raise ~loc
```

Comment:
This also breaks loading the rules, but it's fine for now

Reply:
Is there a rule of thumb to know when it's fine to raise user errors and break the rules loading? I have the impression that my criteria is generally more lenient (e.g. #6412 was fine to fail early), but I don't get the big picture yet.